File: src/py_pkg/py_pkg/talker.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
 
class Talker(Node):
    def __init__(self):
        super().__init__("talker") 
        self.get_logger().info("Talker has been started.")
        self._publisher = self.create_publisher(String, "sc_robo", 10)
        self._delay = 1.0
        self._timer = self.create_timer(self._delay, self.talk)

    def talk(self):
        msg = String()
        msg.data = "Hello, talker from ROS2 Python here."
        self._publisher.publish(msg)
 
def main(args=None):
    rclpy.init(args=args)
    node = Talker() 
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()

File: src/cpp_pkg/src/talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class Listener : public rclcpp::Node 
{
public:
    Listener() : Node("cpp_talker") 
    {
        publisher = this->create_publisher<example_interfaces::msg::String>("sc_robo", 10);
        timer = this->create_wall_timer(1s, std::bind(&Listener::talk, this));
        RCLCPP_INFO(this->get_logger(), "C++ talker has been started.");
    }
 
private:
    void talk()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hello, talker from ROS2 C++ here!");
        publisher->publish(msg);
    }

    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
    rclcpp::TimerBase::SharedPtr timer;
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<Listener>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}